src/Array.mo

```diff
@@ -162,18 +162,18 @@ module {
   };
 
   /// Create a new array by appending the values of `array1` and `array2`.
-  /// @deprecated `Array.append` copies its arguments and has linear complexity;
+  /// Note that `Array.concat` copies its arguments and has linear complexity;
```

Comment:
Drive-by comment: In Haskell `concat` has the signature `[[T]] -> [T]` maybe we should consider this here.

Reply:
Seems reasonable to me; same with Rust and Python.

Reply:
Ditto SML and Caml. Seems a shame to add to the confusion. 

Two options: 
* we just remove the deprecation from append, which, IMO, was poorly motivated in the first place: not all uses are bad.
* we call the new one something other that `concat` - `conjoin`?

Reply:
Or even just `append_`...